### PR TITLE
[fix]: [CDS-35447]: passing server deployment resource selector in non-canar…

### PR DESCRIPTION
**Pull request by tchaurasiya**
Please do not merge it, it has bug
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31758)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingBaseHandler.java [hunk 1]
```diff
@@ -88,12 +88,18 @@ public void setCustomWorkloadsInRelease(List<KubernetesResource> customWorkloads
     }
   }
 
-  public void addLabelsInDeploymentSelectorForCanary(
-      boolean inCanaryWorkflow, List<KubernetesResource> managedWorkloads) {
-    if (!inCanaryWorkflow || isEmpty(managedWorkloads)) {
+  public void addLabelsInDeploymentSelectorForCanary(boolean inCanaryWorkflow,
+      List<KubernetesResource> managedWorkloads, boolean skipAddingTrackSelectorToDeployment) {
+    if (!(inCanaryWorkflow || skipAddingTrackSelectorToDeployment)) {
```
Comment (shubham-in10se) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingBaseHandler.java:93]: wouldn't it be more readable like: `!inCanaryWorkflow && !skipAddingTrackSelectorToDeployment`

**Review by shubham-in10se — APPROVED** [review]

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31758)* status: 2 file(s) to review. 1. unresolved discussions. 

___
*[930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingBaseHandler.java, line 93 at r1](https://reviewable.io/reviews/harness/harness-core/31758#-Mz-NzD4gq308QJ1dOJG-r1-93:-N--1NkV38N0EBSEeJf0:b2ngthx) ([raw file](https://github.com/harness/harness-core/blob/2666ab1de32ce082bf7fa5e613d2a542c25127aa/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingBaseHandler.java#L93)):*
<details><summary><i>Previously, shubham-in10se (Shubham Maheshwari) wrote…</i></summary><blockquote>

wouldn't it be more readable like: `!inCanaryWorkflow && !skipAddingTrackSelectorToDeployment`
</blockquote></details>

+1

**Comment by tchaurasiya** [comment]
*[930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingBaseHandler.java, line 93 at r1](https://reviewable.io/reviews/harness/harness-core/31758#-Mz-NzD4gq308QJ1dOJG-r1-93:-N-1gJCHEhArqtHTAYsW:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2666ab1de32ce082bf7fa5e613d2a542c25127aa/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingBaseHandler.java#L93)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

+1 
</blockquote></details>

Done.

**Review by abosii-harness — DISMISSED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31758)* status: 1. unresolved discussions.

**Review by shubham-in10se — DISMISSED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31758)* status: :shipit: complete! ✓

**Comment by tchaurasiya** [comment]
trigger ti2

**Review by abosii-harness — DISMISSED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31758)* status: :shipit: complete! ✓

**Review by abosii-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31758)* status: :shipit: complete! ✓

**Pull request merged**